src/solvers/dg/dg_2d.jl

```diff
@@ -363,61 +367,188 @@ function calc_volume_integral!(du::AbstractArray{<:Any,4}, u, nonconservative_te
-    fv_kernel!(du, u, equations, volume_flux_fv, dg, cache, element, alpha_element)
+    fv_kernel!(du, u, nonconservative_terms, equations, volume_flux_fv, dg, cache, element, alpha_element)
   end
 
   return nothing
 end
 
-@inline function fv_kernel!(du::AbstractArray{<:Any,4}, u::AbstractArray{<:Any,4},
+@inline function fv_kernel!(du::AbstractArray{<:Any,4}, u::AbstractArray{<:Any,4}, nonconservative_terms,
                             equations, volume_flux_fv, dg::DGSEM, cache, element, alpha=true)
-  @unpack fstar1_threaded, fstar2_threaded = cache
+  @unpack fstar1_L_threaded, fstar1_R_threaded, fstar2_L_threaded, fstar2_R_threaded = cache
   @unpack inverse_weights = dg.basis
 
   # Calculate FV two-point fluxes
-  fstar1 = fstar1_threaded[Threads.threadid()]
-  fstar2 = fstar2_threaded[Threads.threadid()]
-  calcflux_fv!(fstar1, fstar2, u, equations, volume_flux_fv, dg, element)
+  fstar1_L = fstar1_L_threaded[Threads.threadid()]
+  fstar2_L = fstar2_L_threaded[Threads.threadid()]
+  fstar1_R = fstar1_R_threaded[Threads.threadid()]
+  fstar2_R = fstar2_R_threaded[Threads.threadid()]
+  calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u,
+               nonconservative_terms, equations, volume_flux_fv, dg, element)
 
   # Calculate FV volume integral contribution
   for j in eachnode(dg), i in eachnode(dg)
     for v in eachvariable(equations)
       du[v, i, j, element] += ( alpha *
-                                (inverse_weights[i] * (fstar1[v, i+1, j] - fstar1[v, i, j]) +
-                                 inverse_weights[j] * (fstar2[v, i, j+1] - fstar2[v, i, j])) )
-
+                                (inverse_weights[i] * (fstar1_L[v, i+1, j] - fstar1_R[v, i, j]) +
+                                 inverse_weights[j] * (fstar2_L[v, i, j+1] - fstar2_R[v, i, j])) )
     end
   end
 
   return nothing
 end
 
-@inline function calcflux_fv!(fstar1, fstar2, u::AbstractArray{<:Any,4},
-                              equations, volume_flux_fv, dg::DGSEM, element)
 
-  fstar1[:, 1,            :] .= zero(eltype(fstar1))
-  fstar1[:, nnodes(dg)+1, :] .= zero(eltype(fstar1))
+"""
+    calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u_leftright,
+                 nonconservative_terms::Val{false}, equations, volume_flux_fv, dg, element_id)
+
+Calculate the finite volume fluxes inside the elements (without non-conservative terms).
+
+# Arguments
+- `fstar1_L::AbstractArray{T} where T<:Real`:
+- `fstar1_R::AbstractArray{T} where T<:Real`:
+- `fstar2_L::AbstractArray{T} where T<:Real`:
+- `fstar2_R::AbstractArray{T} where T<:Real`:
+- `u_leftright::AbstractArray{T} where T<:Real`
+- `nonconservative_terms::Logical`
+- `equations`
+- `volume_flux_fv`
+- `dg::DGSEM`
+- `element_id::Integer`
+"""
+@inline function calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u::AbstractArray{<:Any,4},
+                              nonconservative_terms::Val{false}, equations, volume_flux_fv,
+                              dg::DGSEM, element)
```

Comment:
```suggestion
                              dg::DGSEM, element_id)
```
Consistency with comment

Reply:
I would rather do it the other way and adapt the comment above.

Reply:
In Taam we consistently used `element_id` for passing the id of an element. Did you change this in Taal :scream_cat: If yes, then I agree with your comment...

Reply:
Yeah, I introduced the function `eachelement` to write `for element in eachelement(...)`, similarly to `for i in eachnode(dg)`. There, I didn't like `eachelement_id` or `each_element_id` but wanted to mimic `eachindex` from `Base`. Hence, `for element_id in eachelement(...)` doesn't look consistent to me and I used `element` instead of `element_id` in Taal (in most places).